src/main/java/net/querz/mcaselector/io/mca/Chunk.java

```diff
@@ -82,39 +84,40 @@ public void load(RandomAccessFile raf) throws IOException {
 		}
 	}
 
-	public int save(RandomAccessFile raf) throws IOException {
-		ExposedByteArrayOutputStream baos = null;
+	public int save(DataOutput out) throws IOException {
+		if (data == null) {
+			throw new IllegalStateException("Can't save unloaded chunk");
+		}
 
-		DataOutputStream nbtOut = switch (compressionType) {
-			case GZIP, GZIP_EXT -> new DataOutputStream(new BufferedOutputStream(new GZIPOutputStream(baos = new ExposedByteArrayOutputStream())));
-			case ZLIB, ZLIB_EXT -> new DataOutputStream(new BufferedOutputStream(new DeflaterOutputStream(baos = new ExposedByteArrayOutputStream())));
-			case NONE, NONE_EXT -> new DataOutputStream(new BufferedOutputStream(baos = new ExposedByteArrayOutputStream()));
-		};
+		ExposedByteArrayOutputStream baos = new ExposedByteArrayOutputStream();
+
+		// CHECK DataOutputStream wrapper unnecessary?
```

Comment:
Both wrappers are unnecessary, because [NBTWriter#write] will wrap with a `BufferedOutputStream` and a `DataOutputStream`.
So this should look like this:
```java
OutputStream nbtOut = switch (compressionType) {
	case GZIP, GZIP_EXT -> new GZIPOutputStream(baos);
	case ZLIB, ZLIB_EXT -> new DeflaterOutputStream(baos);
	case NONE, NONE_EXT -> baos;
};
```

Reply:
Same for both of the `Chunk#load` methods.